packages/react-server/core/util/ClientCssHelper.js

```diff
@@ -29,49 +31,51 @@ module.exports = {
 			throw new Error("ClientCssHelper.registerPageLoad can't be called server-side");
 		}
 
-		var newCss = pageObject.getHeadStylesheets();
+		return Q.all(PageUtil.standardizeStyles(pageObject.getHeadStylesheets())).then(newCss => {
```

Comment:
Seems like these should be chained?  We want to get the _first_ stylesheet `<link/>` tags down to the browser ASAP.  If _later_ stylesheet promises are slower they could delay _earlier_ writes to the browser.